### DISCUSSION TITLE
Does destructuring hits performance?

### DISCUSSION THREAD

**Discussion by fernandoeeu** [pmndrs/zustand#913] [Q&A] [answered]
Are

```const name = useUserState(state => state.name)```

and

```const { name } = useUserState(state => state)```

the same?
Or is the latter the same as ```const state = useUserState()``` as [quoted here](https://github.com/pmndrs/zustand#fetching-everything)

**Comment by dai-shi** [comment, marked as answer]
> is the latter the same as `const state = useUserState()` as [quoted here](https://github.com/pmndrs/zustand#fetching-everything)

Yes.

In case you missed it, we have [valtio](https://valtio.pmnd.rs) that automatically optimizes with destructuring. Alternatively, [react-tracked supports zustand](https://react-tracked.js.org/docs/tutorial-zustand-01/) too.

**Reply by fernandoeeu** [reply]
I've read about it on your twitter profile, but didn't check. Gonna take a look, even though I think Jotai can handle pretty well my needs! Thanks 🤝
